lib/srv/db/common/auth.go

```diff
@@ -52,6 +52,10 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// azureVirutalMachineCacheTTL is the default TTL for Azure virtual machine
+// cache entries.
+const azureVirutalMachineCacheTTL = 5 * time.Minute
```

Comment:
We don't just call the metadata server; we also call the Azure VM endpoint. I think the cache is good to have because the MSSQL clients usually create multiple connections in a short period (tested with Azure Data Studio and mssql-cli). Having it cached can improve a lot the time to connect. That said, we can reduce the cache TTL to reduce the time to propagate the changes.